lib/internal/bootstrap/node.js

```diff
@@ -23,11 +23,58 @@
                                 _memoryUsage, _rawDebug,
                                 _umask, _initgroups, _setegid, _seteuid,
                                 _setgid, _setuid, _setgroups,
-                                _shouldAbortOnUncaughtToggle },
+                                _shouldAbortOnUncaughtToggle,
+                                _setupPrepareStackTrace },
                               { internalBinding, NativeModule },
                               triggerFatalException) {
   const exceptionHandlerState = { captureFn: null };
   const isMainThread = internalBinding('worker').threadId === 0;
+  const { safeToString } = internalBinding('util');
+
+  let wrapper;
+  _setupPrepareStackTrace((globalThis, error, trace, message) => {
+    // Polyfill of V8's Error.prepareStackTrace API.
+    // By owning this call, we can prevent error double-decoration.
+    // TODO(devsnek): swap to `globalThis.Error`
+    if (typeof Error.prepareStackTrace === 'function') {
+      return Error.prepareStackTrace(error, trace);
+    }
+
+    const errorString = safeToString(error);
+    let [
+      startPosition,
+      endPosition,
+      sourceLine,
+    ] = message;
+
+    // The default formatting that V8 uses is:
+    // `${errorString}\n    at ${trace.join('\n    at ')}`;
+    //
+    // We use the same format, but with information about the throw site
+    // prepended in front of it:
+    //
+    //   return 5 aaaaa
+    //            ^^^^^
+    // SyntaxError: Unexpected identifier
+    //     at new Script (vm.js:80:7)
+    //     at createScript (vm.js:264:10)
+    //     ...
+
+    if (wrapper === undefined) {
+      wrapper = NativeModule.require('module').wrapper[0];
```

Comment:
Does this work with native modules?

Reply:
Ideally the PR that uses `compileFunction` will have landed and I can remove this entirely, right now it's kinda just a shim.